plugins/eni/engine/nsclosure.go

```diff
@@ -143,6 +150,22 @@ func (closureContext *setupNamespaceClosureContext) run(_ ns.NetNS) error {
 			"setupNamespaceClosure engine: unable to bring up the device")
 	}
 
+	// Add a blackhole route for IMDS endpoint if required
+	if closureContext.blockIMDS {
+		_, imdsNetwork, err := net.ParseCIDR(instanceMetadataEndpoint)
+		if err != nil {
+			// This should never happen because we always expect
+			// 169.254.169.254/32 to be parsed without any errors
+			return errors.Wrapf(err, "setupNamespaceClosure engine: unable to parse instance metadata endpoint")
+		}
+		if err = closureContext.netLink.RouteAdd(&netlink.Route{
```

Comment:
Just want to understand here, is this like: `ip route add blackhole 169.254.169.254/32`? What's the difference with `ip rule add blackhole to 169.254.169.254/32`?